lib/pacemaker/pcmk_sched_notif.c

```diff
@@ -782,35 +782,39 @@ find_remote_start(pe_action_t *action)
     return NULL;
 }
 
-void
-create_notifications(pe_resource_t * rsc, notify_data_t * n_data, pe_working_set_t * data_set)
+/*!
+ * \internal
+ * \brief Create notify actions, and add notify data to original actions
+ *
+ * \param[in] rsc       Clone or clone instance that notification is for
+ * \param[in] n_data    Clone notification data for some action
+ * \param[in] data_set  Cluster working set
+ */
+static void
+create_notify_actions(pe_resource_t *rsc, notify_data_t *n_data,
+                      pe_working_set_t *data_set)
 {
-    GList *gIter = NULL;
+    GList *iter = NULL;
     pe_action_t *stop = NULL;
     pe_action_t *start = NULL;
     enum action_tasks task = text2task(n_data->action);
 
-    if (rsc->children) {
-        gIter = rsc->children;
-        for (; gIter != NULL; gIter = gIter->next) {
-            pe_resource_t *child = (pe_resource_t *) gIter->data;
+    // If this is a clone, call recursively for each instance
+    if (rsc->children != NULL) {
+        for (iter = rsc->children; iter != NULL; iter = iter->next) {
+            pe_resource_t *child = (pe_resource_t *) iter->data;
 
-            create_notifications(child, n_data, data_set);
+            create_notify_actions(child, n_data, data_set);
```

Comment:
If rsc->cluster could be used, this for loop could then be reduced to `g_list_foreach(rsc->children, (GFunc) create_notify_actions, n_data)`.  This would apply to the same loop as in 3531c8a0550ed4cbce8f664611400e7a744e1c06.

Reply:
That makes sense